wai-aria/role/invalid-roles.html

```diff
@@ -11,26 +11,109 @@
+<div role="foo&#771" data-testname="div with role foo that has tilde diacritic" class="ex-generic">x</div>
+<div role="button&#771" data-testname="div with role set to button with tilde diacritic" class="ex-generic">x</div>
+
+<!-- Zero-width joiners (e.g., ZWJ like emoji variants use) with <button> -->
+<button role="&#8205" data-testname="button with role set to zero-width joiner" data-expectedrole="button" class="ex">x</button>
+<button role="foo&#8205" data-testname="button with role set to foo with zero-width joiner" data-expectedrole="button" class="ex">x</button>
+
+<!-- Non-western chars with <nav> -->
+<nav role="Θ" data-testname="nav with role set to theta (Θ)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="ژ" data-testname="nav with role set to Persian character (ژ)" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="ژ ♮" data-testname="nav with multiple non-latin character roles, Persian character (ژ) and ♮" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="漢字" data-testname="nav with role set to Japanese kanji" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- Non-western chars with <span> -->
+<span role="Θ" data-testname="span with role set to theta (Θ)" class="ex-generic">x</span>
+<span role="ژ" data-testname="span with role set to Persian character (ژ)" class="ex-generic">x</span>
+<span role="ژ ♮" data-testname="span with multiple non-latin character roles, Persian character (ژ) and ♮" class="ex-generic">x</span>
+<span role="漢字" data-testname="span with role set to Japanese kanji" class="ex-generic">x</span>
+
+<!-- RTL strings (Hebrew & Arabic) with <nav> -->
+<nav role="سلام" data-testname="nav with role set to Arabic text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="سلام حبيب" data-testname="nav with multiple role assignments set to Arabic text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="שלום" data-testname="nav with role set to Hebrew text" data-expectedrole="navigation" class="ex">x</nav>
+<nav role="שלום חבר" data-testname="nav with multiple role assignments set to Hebrew text" data-expectedrole="navigation" class="ex">x</nav>
+
+<!-- RTL strings (Hebrew & Arabic) with <div> -->
+<div role="سلام" data-testname="div with role set to Arabic text" class="ex-generic">x</div>
+<div role="سلام حبيب" data-testname="div with multiple role assignments set to Arabic text" class="ex-generic">x</div>
+<div role="שלום" data-testname="div with role set to Hebrew text" class="ex-generic">x</div>
+<div role="שלום חבר" data-testname="div with multiple role assignments set to Hebrew text" class="ex-generic">x</div>
+
+<!-- Escaped chars, URL-encoded chars with <a> -->
+<a role="&amp" data-testname="link with role set to ampersand character" data-expectedrole="navigation" class="ex">x</a>
+<a role="&lt" data-testname="link with role set to less than angle bracket character" data-expectedrole="navigation" class="ex">x</a>
```

Comment:
The `data-expectedrole="navigation"` seems wrong on these, too. The implementations seem to agree these should be generic/none/empty.

Reply:
@cookiecrook Added `href`s for these as well but still curious about role calculation when `href` is omitted (if spec states these should be generic/none, I don't believe I'm observing that currently with Safari/Chrome).

Reply:
That's tested in https://github.com/web-platform-tests/wpt/blob/master/html-aam/roles-contextual.html#L20

Reply:
but in either case, the role would not be "navigation"…

Reply:
diff here https://github.com/web-platform-tests/wpt/pull/42013#discussion_r1349410042

Reply:
Fixed in b3f7a8ad452c1729168f7a72751ba84665b44f3d.